pkg/federate/federator.go

```diff
@@ -4,39 +4,17 @@ package federate
 
 import (
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/client-go/rest"
 )
 
-// ClusterEventHandler can handle notifications for events that happen to a federated cluster.
-type ClusterEventHandler interface {
-	// OnAdd is called when a cluster is added. The given 'kubeConfig' can be used to access
-	// the cluster's kube API endpoint.
-
-	OnAdd(clusterID string, kubeConfig *rest.Config)
-
-	// OnUpdate is called when some aspect of a cluster's kube API endpoint configuration has changed.
-	OnUpdate(clusterID string, kubeConfig *rest.Config)
-
-	// OnRemove is called when a cluster is removed.
-	OnRemove(clusterID string)
-}
-
 // Federator provides methods for accessing federated clusters.
 type Federator interface {
-	// WatchClusters adds a ClusterEventHandler to be notified when cluster events occur.
-	// The handler is notified asynchronously of the existing set of clusters via OnAdd events, one per cluster.
-	WatchClusters(handler ClusterEventHandler) error
-
-	// Distribute distributes the given resource to the given list of clusters referred to by their IDs.
+	// Distribute distributes the given resource to all federated clusters.
 	// The actual distribution may occur asynchronously in which case any returned error only indicates that the request
 	// failed.
 	//
-	// If clusterNames is not passed, the resource will be distributed to all clusters
-	//
 	// If the resource was previously distributed and the given resource differs, each previous cluster will receive the
-	// updated resource. If previously distributed and the given list of clusters differs, the resource will be
-	// distributed to any new clusters in the updated list and deleted from previous clusters not in the updated list.
-	Distribute(resource runtime.Object, clusterNames ...string) error
+	// updated resource.
+	Distribute(resource runtime.Object) error
```

Comment:
Theoretically we could limit the spread of information only to specific clusters even with a broker model, but I agree that it's better to remove if we don't use this anyway.
Only thing I might've changed is the method name to `DistributeOnAllClusters` or something of this sort, since go doesn't have method overloading.

Reply:
yeah I'd rather not include a feature in the API that won't be implemented or used going forward (although it wasn't used before anyway). We can always add it back later if needed.

The fact that it distributes to all clusters is mentioned in the doc comments, thus I don't think the name needs to specifically indicate that for brevity.